### PR TITLE
Trigger main when scipp main completes

### PR DESCRIPTION
**Pull request by OwenArnold**
scipp **project**, Main **pipeline** is upstream trigger.

Help identify upstream builds that break things in scipp-neutron. This should be replaced/incorporated into the wider CI overhaul underway by @nvaytet 

**Tester**

It is possible to test these changes prior to merging, but it's convoluted, not worth the effort IMHO.

### CODE REVIEW

File: .azure-pipelines/main.yml [hunk 1]
```diff
@@ -3,6 +3,12 @@ trigger:
 
 pr: none
 
+resources:
+  pipelines:
+  - pipeline: upstream_scipp_release
```
Comment (SimonHeybrock) [inline .azure-pipelines/main.yml:8]: Release?

**Review by SimonHeybrock — COMMENTED** [review]
Is this triggering a new package build? Should it really? I suppose we are interested in two things:

- Does `scippneutron` still build and pass tests.
- Does the existing `scippneutron` conda package still work with the new uptream package?

It feels oddd to publish a new package for no reason.

**Comment by OwenArnold** [comment]
> Is this triggering a new package build? Should it really? I suppose we are interested in two things:
> 
> * Does `scippneutron` still build and pass tests.
> * Does the existing `scippneutron` conda package still work with the new uptream package?
> 
> It feels oddd to publish a new package for no reason.

Maybe this is pointing at the need for a new pipleline type to handle this. I agree there is no need to build and publish. I especially like the idea of not having to build to check ABI compatibility.

File: .azure-pipelines/main.yml (hunk 1 above)
Comment (SimonHeybrock) [reply .azure-pipelines/main.yml:8]:
```suggestion
  - pipeline: upstream_scipp_main
```

**Comment by SimonHeybrock** [comment]
The new pipeline discussed is maybe the way to go long term, but for now this is simple and should do the job, so I merged this.

**Pull request merged**